NebulaPatcher/Patches/Dynamic/DESelection_Patch.cs

```diff
@@ -30,35 +30,37 @@ public static bool SetViewStar_Prefix(DESelection __instance, ref StarData starD
                                 return true;
                             }
                         }
+                        //No dyson sphere exist, close the UI
+                        UIRoot.instance.uiGame.dysonEditor._Close();
+                        return false;
+                    }
+                    else if (GameMain.data.dysonSpheres[starData.index] == null)
+                    {
+                        //Local dyson sphere hasn't loaded yet, close the UI
+                        UIRoot.instance.uiGame.dysonEditor._Close();
+                        return false;
                     }
                 }
                 if (starData != null && GameMain.data.dysonSpheres[starData.index] == null)
                 {
+                    if (Multiplayer.Session.DysonSpheres.RequestingIndex != -1)
+                    {
+                        InGamePopup.ShowInfo("Loading", $"Loading Dyson sphere {starData.displayName}, please wait...", "OK");
```

Comment:
As you call fadeout later, why add the Ok button here? :)

Reply:
some users may be impatient about loading time, so I add a way to let them close the info window. (although Esc can do the same thing)